src/app/shared/conditional-show/conditional-show.directive.ts

```diff
@@ -116,4 +129,33 @@ export class ConditionalShowDirective implements AfterViewInit, OnDestroy {
       this.formChangesSubscription.unsubscribe();
     }
   }
+
+  private checkHideShowCondition(key: string, aControl: AbstractControl) {
```

Comment:
Just a thought but was wondering how difficult would it be to make this a separate formControlTraverser passing in a condition of the control (in this case `if (aControl.invalid)`) and action to take once condition is met (in this case `this.registry.refresh()`).

Reply:
Created https://tools.hmcts.net/jira/browse/RDM-2233 for this purpose...